extensions/registries/src/extension.ts

```diff
@@ -0,0 +1,66 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as extensionApi from '@tmpwip/extension-api';
+
+export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
+  // Create the "provider"
+  const provider = extensionApi.provider.createProvider({
+    name: 'Registries',
+    id: 'registries',
+    status: 'unknown',
+    images: {
+      icon: './icon.png',
+      logo: './icon.png',
+    },
+  });
+  extensionContext.subscriptions.push(provider);
+
+  // For each defaultRegistries, suggest the registry to Podman Desktop
+  defaultRegistries.forEach(registry => {
+    extensionApi.registry.suggestRegistry(registry);
+  });
+}
+
+// Const array of list of approved registries that contain the default URL as well as base64 encoded version of their logo
+// Name
+// URL
+// Base64 png of logo (optional), no need to use `fs` filesystem to load + convert if just providing a small base64 image.
+// each image is a 40x40px base64 encoded png
+const defaultRegistries: extensionApi.RegistrySuggestedProvider[] = [
+  {
+    name: 'Docker Hub',
+    url: 'docker.io',
+    icon: 'iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAABhmlDQ1BJQ0MgcHJvZmlsZQAAKJF9kT1Iw1AUhU9TpSIVwXYQcchQnSyIioguWoUiVAi1QqsOJi/9gyYNSYqLo+BacPBnserg4qyrg6sgCP6AuLo4KbpIifclhRaxXni8j/PuObx3HyDUSkyzOkYBTbfNZDwmpjOrYuAVPvQhhGnMyMwy5iQpgbb1dU/dVHdRntW+78/qUbMWA3wi8SwzTJt4g3hy0zY47xOHWUFWic+JR0y6IPEj1xWP3zjnXRZ4ZthMJeeJw8RivoWVFmYFUyOeII6omk75QtpjlfMWZ61UYY178hcGs/rKMtdpDSKORSxBgggFFRRRgo0o7TopFpJ0HmvjH3D9ErkUchXByLGAMjTIrh/8D37P1sqNj3lJwRjQ+eI4H0NAYBeoVx3n+9hx6ieA/xm40pv+cg2Y+iS92tQiR0DvNnBx3dSUPeByB+h/MmRTdiU/LSGXA97P6JsyQOgW6F7z5tY4x+kDkKJZJW6Ag0NgOE/Z623e3dU6t397GvP7AfSXctvjBGIeAAAABmJLR0QAAAAAAAD5Q7t/AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH5wESFAAdeMxQSwAAAeNJREFUSMftlj9I1VEUxz9PnosQDjcKhFDkDoJbtQji7BBEDS0ilKJ2dXISGtwSl0QHvRTq0FBE4BZFgvgHdJBcXrl4EwRdzEsU5KQ9l/Pg+uNd8D1+P4J4Z/mdwzm/3/fc8/ueLxdq9r9brpqXtPWDwEsJF51R/ZV+o+5fNZ+vEswDX8U/TBVYWz8KtEn4ChgUvwi8BjYldmmfuAe4I/5OAJyM66WRcs1PATPOqIMs/nFsYveBUWBNW38zemJtfS5opAiMA80SfwKMEOkcWAWGJberra8LSPbXGVUEHkrcDEwCT2KjHghW5CNwIxj1U8AGtUPAXGmdgBdB7ZBwoj2of6Ct/wxcAwrOqK18CmsWq/0eNNMIvAmmewn4BPgi/jfgepA7CnLJ2n3gLMgdy/M98CjS1GpOW/8Y6EiZW7eAThlt0uaBkTwwCzRkLM3vgA1g3RlVKJHrGGjJEPQE6HNGnSaJ4TM+7VwStARcyBDUA1OxVVjLEHjMGfUrBvxB1ChtW3JGLUSX3xn1A3ibMugG0HsV1XkG/EkJdAXoLkeosrcHbX2XaHS1O30KPHdGTVSks86odaAL2K4Q8CcwDbReFTR6X9LW3wXuAbeBJkCL0P8G9kQUduRfLjujzmr35pol7QICy39acx6DywAAAABJRU5ErkJggg==',
```

Comment:
We should read from files stored locally as it's easier to maintain

Reply:
Yeah, I was debating about doing it via icon.png vs base64, since the icons were so small. I'll load it locally instead!